pytorch_pfn_extras/training/extensions/slack.py

```diff
@@ -0,0 +1,300 @@
+import getpass
+import os
+import json
+import urllib.request
+import shlex
+import sys
+import socket
+import warnings
+
+from typing import Any, Callable, Dict, List, Optional, Union, TYPE_CHECKING
+
+from pytorch_pfn_extras.training import extension
+from pytorch_pfn_extras.training._manager_protocol import ExtensionsManagerProtocol
+
+
+if TYPE_CHECKING:
+    from pytorch_pfn_extras.training._trigger_util import TriggerLike
+
+
+try:
+    import slack_sdk
+    _slack_sdk_available = True
+except ImportError:
+    _slack_sdk_available = False
+
+
+_identity = f'{getpass.getuser()}@{socket.gethostname()} [PID {os.getpid()}]'
+
+
+def _default_start_msg(m: ExtensionsManagerProtocol, c: Any, o: Dict[Any, Any]) -> str:
+    return f'''**Training started! {_identity}**
+Command: `{shlex.quote(' '.join(sys.argv))}`
+'''
+
+
+_default_end_msg = f'**Training finished! {_identity}**'
+
+
+class Slack(extension.Extension):
+    """An extension to communicate with Slack.
+
+    This extension receives a ``msg`` argument that contains
+    placeholders that will be populated with ``manager`` or ``manager.observation``
+    fields, and custom objects such as ``context_object``.
+
+    For example. `msg = "Loss {loss} for iteration {.iteration}"`
```

Comment:
Please update the example format.

```suggestion
    For example, `msg = "**Epoch #{manager.epoch}**: loss {val/loss}"`
```

The following sentences must be updated as well.